src/lib/core/CHIPConfig.h

```diff
@@ -1856,6 +1856,19 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
 #define CHIP_CONFIG_TEST_GOOGLETEST 0
 #endif // CHIP_CONFIG_TEST_GOOGLETEST
 
+/**
+ * @def CHIP_CONFIG_TC_REQUIRED
+ *
+ * @brief Indicates whether terms and conditions are required during commissioning.
+ *
+ * This macro defines whether the device commissioning process requires the user to acknowledge terms and conditions.
+ * - 1: Terms and conditions are required.
+ * - 0: Terms and conditions are not required.
+ */
+#ifndef CHIP_CONFIG_TC_REQUIRED
+#define CHIP_CONFIG_TC_REQUIRED 0
```

Comment:
do we have a gn build flag to enable this? how does one compile with this feature enabled? Is this only by editing the chipconfig.h?

For others (like googletest above) we have a buildconfig header that allows control for it: https://github.com/project-chip/connectedhomeip/blob/master/src/lib/core/BUILD.gn#L73

we should have one for this as well.

Reply:
Updated this in a followup to make this configurable and part of an app build. Closing the comment.